synapseclient/client.py

```diff
@@ -596,17 +534,49 @@ def get_client(cls, synapse_client: typing.Union[None, "Synapse"]) -> "Synapse":
             An instance of 'Synapse'.
 
         Raises:
-            SynapseError: No instance has been created - Please use login() first
+            SynapseError: No instance has been passed into a function where
+                the `synapse_client` kwarg is available, and no instance has been cached
+                during the Synapse constructor.
         """
         if synapse_client:
             return synapse_client
 
         if not cls._synapse_client:
             raise SynapseError(
-                "No instance has been created - Please use login() first"
+                "No instance has been passed into a function where the `synapse_client` kwarg is available, and no instance has been cached during the Synapse constructor."
             )
         return cls._synapse_client
 
+    @staticmethod
+    def allow_client_caching(allow_client_caching: bool) -> None:
+        """Allows for a global setting to enable or disable caching of the Synapse
+        client object. **If you are working in a multi-user environment you should set
+        this to False to prevent any possibility of picking up another user session.**
+
+
+        As a consequence of setting this to false you must pass a Synapse class instance
+        to any function, method, or class that requires it - If if it is marked as
+        `Optional`.
+        """
+        Synapse._allow_client_caching = allow_client_caching
+
+    @staticmethod
+    def enable_open_telemetry(enable_open_telemetry: bool) -> None:
+        """Determines whether OpenTelemetry is enabled for the Synapse client. This is
+        used to know whether or not this library will automatically kick off the
+        instruementation of several dependent libraries including:
+
+            - threading
+            - urllib
+            - requests
+            - httpx
+        """
+        if enable_open_telemetry and not Synapse._enable_open_telemetry:
```

Comment:
I would add some comments here or in the docstring to clarify the logic a bit more:

```
# Turns OpenTelemetry on when it was previously off
...
# Turns OpenTelemetry off when it was previously on
...
```

Reply:
Added:

        When OpenTelemetry is enabled it will automatically start the instrumentation
        of these libraries. When it is disabled it will automatically stop the
        instrumentation of these libraries.